contracts/governance/ApprovalReceiver.sol

```diff
@@ -0,0 +1,80 @@
+pragma solidity ^0.5.17;
+
+import "./ErrorDecoder.sol";
+import "../token/IApproveAndCall.sol";
+
+/**
+ * Base contract for receiving approval from SOV token
+ */
+contract ApprovalReceiver is ErrorDecoder, IApproveAndCall {
+	modifier onlyThisContract() {
+		//accepts calls only from receiveApproval function
+		require(msg.sender == address(this), "unauthorized");
+		_;
+	}
+
+	/**
+	 * @notice receives approval from SOV token
+	 * @param _data the data will be used for low level call
+	 */
+	function receiveApproval(
+		address, /*_sender*/
+		uint256, /*_amount*/
+		address, /*_token*/
+		bytes memory _data
+	) public {
```

Comment:
`external` - it saves a bit of gas and improves readability

Reply:
done